src/main/java/seedu/address/ui/PersonListPanel.java

```diff
@@ -32,9 +32,10 @@ public class PersonListPanel extends UiPart<Region> {
     public PersonListPanel(Logic logic) {
         super(FXML);
         this.logic = logic;
-        this.reRenderUi = logic.getRefreshListUi();
+        this.reRenderUi = logic.getRefreshListUi(); // Connect to the logic manager's boolean flag
         personListView.setItems(logic.getFilteredPersonList());
         personListView.setCellFactory(listView -> new PersonListViewCell());
+        // Listens for a change in reRenderUi value and updates UI, the actual value is irrelevant
         reRenderUi.addListener(new ChangeListener<Boolean>() {
```

Comment:
Just for clarification, the reason why this is needed is because the ListView is not auto updating when setting the Display Fields?

Reply:
yup, changing the displayedFieldsList values doesnt update the UI, hence we need a listener to track when to call for refresh

Reply:
Alright noted, thanks!